src/services/runtime/RuntimeMetadataService.ts

```diff
@@ -32,4 +34,45 @@ export class RuntimeMetadataService extends AbstractService {
 
 		return metadata;
 	}
+
+	/**
+	 * Fetch the requested version of `Metadata` in decoded JSON form.
+	 *
+	 * @param hash `BlockHash` to make call at
+	 */
+	async fetchMetadataVersioned(hash: BlockHash, metadataVersion: number): Promise<Metadata> {
+		const { api } = this;
+		const apiAt = await api.at(hash);
+
+		let metadata: Option<OpaqueMetadata> | undefined;
+		let metadataVersioned: Metadata | undefined;
+		try {
+			metadata = await apiAt.call.metadata.metadataAtVersion(metadataVersion);
+			if (metadata) {
+				metadataVersioned = new Metadata(apiAt.registry, metadata.unwrap());
+			} else {
+				throw new Error(`Metadata for version ${metadataVersion} is not available.`);
+			}
+		} catch {
+			throw new InternalServerError(
+				`An error occured while attempting to fetch ${metadataVersion.toString()} metadata.`,
+			);
+		}
+
+		return metadataVersioned;
+	}
+
+	/**
+	 * Fetch the available `Metadata` versions.
+	 *
+	 * @param hash `BlockHash` to make call at
+	 */
+	async fetchMetadataVersions(hash: BlockHash): Promise<string[]> {
+		const { api } = this;
+		const apiAt = await api.at(hash);
```

Comment:
Same with this api.

Reply:
I changed the previous one but for this one, in the `RuntimeMetadataController` > `getMetadataAvailableVersions` I don't call `api.at`. I just get the hash.